source/solutions-library/open-finance-data-store.txt

```diff
@@ -0,0 +1,254 @@
+approach leverages MongoDB's flexible document model to handle diverse data structures
+from different financial entities. Here’s a detailed description of the data model
+approach:
+
+**Simplified components for demonstration purposes**
+
+- **Tokens collection:** 
+  
+  - **Purpose**: Stores Bearer tokens used for authenticating and authorizing API requests
+    between Leafy Bank and external banks. 
+  
+  - **Structure**:
+  
+    .. code-block:: json
+        
+        {
+        "_id": { "$oid": "unique_id" }, 
+        "UserName": "string", 
+        "BearerToken": "string",
+        "TokenDates": {
+            "CreationDate": { "$date": "ISODate" }, 
+            "LastUseDate": { "$date": "ISODate" }
+          },
+        }
+    
+  - **Simplification**: In a real-world scenario, token management would involve more
+    complex mechanisms, including token expiration, refresh tokens, and secure storage
+    practices.
+
+- **External accounts collection:** 
+
+  - **Purpose**: Stores information about external bank accounts linked by users. 
+  
+  - **Structure**:
+  
+    .. code-block:: json
+        
+        {
+        "_id": { "$oid": "unique_id" }, 
+        "AccountId": "string", 
+        "AccountBank": "string",
+        "AccountStatus": "string", 
+        "AccountType": "string", 
+        "AccountBalance": "number",
+        "AccountCurrency": "string", 
+        "AccountDate": {
+            "OpeningDate": { "$date": "ISODate" }
+        }, 
+        "AccountUser": {
+            "UserName": "string", 
+            "UserId": { "$oid": "unique_id" }
+        }, "GreenAccountNarrative": "string"
+        }
+  
+  - **Simplification**: Real-world account data would include more detailed information
+    such as transaction history, account limits, and additional metadata for compliance
+    and auditing.
+
+- **External products collection:** 
+
+  - **Purpose**: Stores information about financial products (e.g., loans and mortgages)
+    associated with users. 
+
+  - **Structure**:
+    
+    .. code-block:: json
+        
+        {
+        "_id": { "$oid": "unique_id" }, 
+        "ProductId": "string", 
+        "ProductBank": "string",
+        "ProductStatus": "string", 
+        "ProductType": "string", 
+        "ProductAmount": "number",
+        "ProductCurrency": "string", 
+        "ProductInterestRate": "number", 
+        "ProductDate": {
+            "OpeningDate": { "$date": "ISODate" }
+        }, 
+        "ProductCustomer": {
+            "UserName": "string", 
+            "UserId": { "$oid": "unique_id" }
+        }, 
+        "GreenProductNarrative": "string", 
+        "RepaymentPeriod": "number", 
+        "LoanCollateral": "string"
+        }
+
+  - **Simplification**: In a real-world scenario, product data would be more
+    comprehensive, including detailed terms and conditions, repayment schedules, and
+    collateral details.
+  - **Real-world scenario considerations**
```

Comment:
This is a correct migration of the article as is, but should probably be a heading in the final version. Nice work on formatting this article in general, I like the cleanup!